src/Areas/Server/Commands/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Reflection;
+using AzureMcp.Areas.Server.Commands.Discovery;
+using AzureMcp.Areas.Server.Commands.Runtime;
+using AzureMcp.Areas.Server.Commands.ToolLoading;
+using AzureMcp.Areas.Server.Options;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ModelContextProtocol.Protocol;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// Extension methods for configuring Azure MCP server services.
+/// </summary>
+public static class AzureMcpServiceCollectionExtensions
+{
+    private const string DefaultServerName = "Azure MCP Server";
+
+    /// <summary>
+    /// Adds the Azure MCP server services to the specified <see cref="IServiceCollection"/>.
+    /// </summary>
+    /// <param name="services">The service collection to add services to.</param>
+    /// <param name="serviceStartOptions">The options for configuring the server.</param>
+    /// <returns>The service collection with MCP server services added.</returns>
+    public static IServiceCollection AddAzureMcpServer(this IServiceCollection services, ServiceStartOptions serviceStartOptions)
+    {
+        // Register options for service start
+        services.AddSingleton(serviceStartOptions);
+        services.AddSingleton(Options.Options.Create(serviceStartOptions));
+
+        // Register tool loader strategies
+        services.AddSingleton<CommandFactoryToolLoader>();
+        services.AddSingleton(sp =>
+        {
+            return new RegistryToolLoader(
+                sp.GetRequiredService<RegistryDiscoveryStrategy>(),
+                sp.GetRequiredService<IOptions<ServiceStartOptions>>(),
+                sp.GetRequiredService<ILogger<RegistryToolLoader>>()
+            );
+        });
+
+        services.AddSingleton<SingleProxyToolLoader>();
+        services.AddSingleton<CompositeToolLoader>();
+        services.AddSingleton<ServerToolLoader>();
+
+        // Register server discovery strategies
+        services.AddSingleton<CommandGroupDiscoveryStrategy>();
+        services.AddSingleton<CompositeDiscoveryStrategy>();
+        services.AddSingleton<RegistryDiscoveryStrategy>();
+
+        // Register server providers
+        services.AddSingleton<CommandGroupServerProvider>();
+        services.AddSingleton<RegistryServerProvider>();
+
+        // Register MCP runtimes
+        services.AddSingleton<IMcpRuntime, McpRuntime>();
+
+        // Configure tool loading based on proxy mode
+        if (serviceStartOptions.Proxy == ProxyModes.Single)
+        {
+            services.AddSingleton<IMcpDiscoveryStrategy>(sp =>
```

Comment:
It looks like this and lines 68 do the exact same thing except one variable is 'toolLoaders' and the other is 'discoveryStrategies" Or am I misreading it.